jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -238,13 +238,12 @@ public InputStream newInputStream() throws IOException
      *
      * @return a readable {@link java.nio.channels.ByteChannel} to the resource or null if one is not available.
      * @throws IOException if unable to open the readable bytechannel for the resource.
+     * @deprecated use {@link #newInputStream()} or {@code IOResources} instead.
      */
+    @Deprecated(since = "12.0.8", forRemoval = true)
```

Comment:
If we do, this means two things:

 - `IOResources` does not have any fallback in case it doesn't know about the resource type.
 - `jetty-util` and `jetty-xml` uses of that method have to go away. Tests can probably safely assume that a resource has a path, but `BufferUtil.toBuffer(Resource, boolean)`, `CertificateUtils`, `XmlConfiguration` and other utilities would need the same kind of logic that `IOResources` has.

Reply:
OK. leave it as is for now.